docs/DeveloperGuide.md

```diff
@@ -99,28 +99,57 @@ Optionally, you can follow the [Using Checkstyle](UsingCheckstyle.md) document t
+ * [`RepoConfiguration`](/src/main/java/reposense/model/RepoConfiguration.java) stores the configuration information from the CSV config file for a single repository, which are the repository's orgarization, name, branch, list of authors to analyse, date range to analyze commits and files from `CliArguments`.
  These configuration information are used by:
     - `GitDownloader` to determine which repository to download from and which branch to check out to.
     - `AuthorshipReporter` and `CommitsReporter` to determine the range of commits and files to analyze.
     - `ReportGenerator` to determine the directory to output the report.
 
 
 ## HTML Dashboard
+The source files for the dashboard is located in `frontend/src` and is built by [spuild](https://github.com/ongspxm/spuild2) before being packaged into the JAR file to be extracted as part of the report.
 
- ![Dashboard](images/dashboard-architecture.png)
+### Javascript Files
+- **api.js** - loading and parsing of the dashboard content
+- **main.js** - main controller that handles the loading of the content into different modules
+- **v_summary.js** - module that supports the ramp chart view
+- **v_authorship.js** - module that supports the authorship view
 
-The dashboard contains two main parts: data `JSONs`(generated by `ReportGenerator`), and the static Dashboard template. Refer to [here](Build.md) for instructions on how to build the project and dashboard.
+### Overall Structure
+The main Vue object (`window.app`) is responsible for the loading of the dashboard (through `summary.json`). Its `repos` attribute is tied to the global `window.REPOS`, and is passed into the various other modules when the information is needed.
 
- * As shown in the graph above, there will be one summary `JSON`(`summary.json`) containing the summary information of all repositories, and one set of repo detail `JSON`(`commits.json` and `authorship.json`) for each repository, containing the contribution information of each specified author to that repository.
+`window.app` is broken down into two main parts, the summary view and the tabs view. The basic design is to have a summary view (with the ramp charts) as the main dashboard, and other modules display additional information in the tabbed interface which is displayed on the right.
 
-  * The dashboard template is a set of `HTML` + `CSS` + `Javascript`, which is located in `template/`. It is used to interpret the `JSON` information and display the dashboard on a web browser. <br>
-  Whenever *RepoSense* is built using *Gradle*, the current dashboard template's content will be zipped into `templateZip.zip` file for easier retrieval, and packaging into the generated `.jar` file.<br>
-  The dashboard template will then be unzipped to the target location along with the generated `JSON` data files in order to produce the dashboard whenever *RepoSense* generates a new report.
+**overall architecture**
+
+![overall architecture](images/dashboard-architecture.png)
```

Comment:
Place this at the top of this section instead, so there's a better flow from an overall of the architecture first, then to the details of each component.
Update the section title to `Dashboard Architecture` as well.